src/docfx/config/ops/OpsAccessor.cs

```diff
@@ -262,6 +249,13 @@ private static HttpMiddleware BuildMiddleware(DocsEnvironment? environment = nul
             };
         }
 
+        private static ValueTask<AccessToken> GetAccessTokenAsync(DocsEnvironment? environment = null)
+        {
+            var defaultAzureCredential = new DefaultAzureCredential();
+            return defaultAzureCredential.GetTokenAsync(
+                new TokenRequestContext(new[] { $"{DocsBuildApiAADClientId(environment)}/.default" }));
```

Comment:
Will a user successfully build a repo if:
- Didn't login to vscode
- Login to azure CLI or visual studio
- Does not have access to build API

Reply:
lemme try

Reply:
@yufeih 
public repo: ✔
private repo: ❌
![image](https://user-images.githubusercontent.com/42199316/129143060-4096f03d-6bef-49e0-b3dc-517fa930c695.png)

@928PJY  Is building private repo as public user a valid use case currently? 
e.g.: Trying to build https://github.com/MicrosoftDocs/test(private), but not an OPS user.

Reply:
I think this is the expected behavior.

Reply:
no, it is not a valid use case for the public user to build a private

Reply:
for building public repo, it's downloading all config related APIs
![image](https://user-images.githubusercontent.com/42199316/129148843-387eb078-add1-4cce-93c2-35db187df143.png)